### PR TITLE
Make unit tests green

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes two remaining unit test failures.

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller_test.go [hunk 1]
```diff
@@ -290,7 +291,13 @@ func Test_NodesDeleted(t *testing.T) {
 				nodeMonitorPeriod: 1 * time.Second,
 			}
 
-			eventBroadcaster.StartLogging(t.Logf)
+			var loggerMutex sync.Mutex
+			logger := func(format string, args ...interface{}) {
+				loggerMutex.Lock()
+				defer loggerMutex.Unlock()
+				t.Logf(format, args...)
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller_test.go:298]: isn't `t.Logf` already thread-safe ?

Comment (p0lyn0mial) [reply staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller_test.go:298]: it looks like it is https://golang.org/src/testing/testing.go

File: pkg/kubelet/kuberuntime/convert_test.go [hunk 2]
```diff
@@ -92,6 +93,12 @@ func TestConvertToKubeContainerImageSpec(t *testing.T) {
 
 	for _, test := range testCases {
 		actual := toKubeContainerImageSpec(test.input)
+		sort.Slice(test.expected.Annotations, func(i, j int) bool {
```
Comment (p0lyn0mial) [inline pkg/kubelet/kuberuntime/convert_test.go:96]: can you sort a map ?

**Comment by marun** [comment]
@mfojtik Please propose to o/k master

**Comment by marun** [comment]
Proposed to master in https://github.com/openshift/kubernetes/pull/298

**Pull request closed without merging**